dot/state/block_funcs.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2019 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package state
+
+import (
+	"bytes"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/common/optional"
+)
+
+// SetReceipt sets a Receipt in the database
+func (bs *BlockState) SetReceipt(hash common.Hash, data []byte) error {
+	bs.lock.Lock()
+	defer bs.lock.Unlock()
+
+	err := bs.db.Put(prefixHash(hash, receiptPrefix), data)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// GetReceipt retrieves a Receipt from the database
+func (bs *BlockState) GetReceipt(hash common.Hash) (*optional.Bytes, error) {
+	data, err := bs.db.Get(prefixHash(hash, receiptPrefix))
+	if err != nil {
+		return nil, err
+	}
+
+	r := &bytes.Buffer{}
+	_, err = r.Write(data)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
what is this? `r` isn't used anywhere

Reply:
some useless garbage it seems, will remove.

Reply:
updated.